### ISSUE TITLE
Feature request: Add "return null for missing leaf" option in SelectTokens

### ISSUE THREAD

**Issue by andremantaswow** [JamesNK/Newtonsoft.Json#2641]
## Feature request

Add an option to allow the return of null for missing leafs when using `SelectTokens`. Maybe an option in https://github.com/JamesNK/Newtonsoft.Json/blob/master/Src/Newtonsoft.Json/Linq/JsonSelectSettings.cs

Example in another lib: https://github.com/json-path/JsonPath#options > DEFAULT_PATH_LEAF_TO_NULL

[Fiddle](https://dotnetfiddle.net/dodbn3)
[Related SO question](https://stackoverflow.com/questions/70748002/newtonsoft-json-how-to-get-null-for-missing-leafs-in-selecttokens)

Example of current output using Newtonsoft

```
var json = @"{ 
  ""Values"": [
    {
      ""Name"": ""Bob"",
      ""Age"": ""25"",
      ""Address"": ""1234 Easy St.""
    },
    {
      ""Age"": ""28"",
      ""Address"": ""1600 Pennsylvania Ave.""
    },
    {
      ""Name"": ""Joe"",
      ""Age"": ""31"",
      ""Address"": ""653 28th St NW""
    }
  ]
}";

var data = JToken.Parse(json);

var query = data.SelectTokens("Values[*].Name");

Console.WriteLine(JsonConvert.SerializeObject(query, Formatting.Indented));
```

Outputs
```
[
  "Bob",
  "Joe"
]
```

With the new feature, the method would return
```
[
  "Bob",
  null,
  "Joe"
]
```

**Comment by omni68** [comment]
I agree. We are using json path syntax in Azure Data Factory (Copy Activity > Mapping) to select nested object values and flatten them into columns, and have the same root problem.

**Comment by smddzcy** [comment]
Have the same problem. This should definitely be an option.

**Comment by onurcskun** [comment]
@smddzcy +++++++++

**Comment by ankiga-MSFT** [comment]
Running into the same problem which flattening JSON Array. Without NULLs, the ordering of the fields is lost.
